crates/worker/src/partition/state_machine/mod.rs

```diff
@@ -347,20 +182,17 @@ impl<Codec> StateMachine<Codec> {
     }
 }
 
-pub(crate) struct StateMachineApplyContext<'a, S> {
-    storage: &'a mut S,
+pub(crate) struct StateMachineApplyContext<'a, 'b: 'a> {
```

Comment:
given we have two lifetimes, maybe give them meaningful names?

Also I'm not so sure you need two lifetimes in this `Context` struct, isn't this reference going to live as long as the mut borrow to transaction anyway for the usage of this struct?

Reply:
We need two lifetimes because we need to tell Rust that this exclusive borrow will be shorter than the lifetime of the transaction.